src/components/pager/__internal__/pager-navigation.component.tsx

```diff
@@ -161,24 +165,33 @@ const PagerNavigation = ({
   return (
     <StyledPagerNavigation>
       {!hasOnePage && renderButtonsBeforeCount()}
-      {showPageCount && (
-        <StyledPagerNavInner>
-          <label htmlFor={currentPageId}>
-            <StyledPagerNoSelect>{l.pager.pageX()}</StyledPagerNoSelect>
-          </label>
-          <NumberInput
-            value={currentPage.toString()}
-            data-element="current-page"
-            onChange={handleCurrentPageChange}
-            onBlur={handlePageInputChange}
+      {showPageCount &&
+        (!nonInteractivePager ? (
+          <StyledPagerNavInner>
+            <label htmlFor={currentPageId}>
+              <StyledPagerNoSelect>{l.pager.pageX()}</StyledPagerNoSelect>
+            </label>
+            <NumberInput
+              value={currentPage.toString()}
+              data-element="current-page"
+              onChange={handleCurrentPageChange}
+              onBlur={handlePageInputChange}
+              id={currentPageId}
+              onKeyUp={(ev) =>
+                Events.isEnterKey(ev) ? handlePageInputChange(ev) : false
+              }
+            />
+            <StyledPagerNoSelect>{l.pager.ofY(pageCount)}</StyledPagerNoSelect>
+          </StyledPagerNavInner>
+        ) : (
+          <StyledPagerNavLabel
+            data-element="current-page-label"
             id={currentPageId}
-            onKeyUp={(ev) =>
-              Events.isEnterKey(ev) ? handlePageInputChange(ev) : false
-            }
-          />
-          <StyledPagerNoSelect>{l.pager.ofY(pageCount)}</StyledPagerNoSelect>
-        </StyledPagerNavInner>
-      )}
+            aria-live="polite"
+          >
+            {l.pager.pageXOfY(currentPage.toString(), pageCount)}
```

Comment:
I don't think we need the extra function as we can use the existing ones to build the string

```suggestion
           {l.pager.pageX()} {currentPage} {l.pager.ofY(pageCount)}
```